starknet_py/net/models/transaction.py

```diff
@@ -58,7 +58,7 @@ def compute_invoke_hash(
         contract_address=contract_address,
         entry_point_selector=entry_point_selector,
         calldata=calldata,
-        chain_id=chain_id.value,
+        chain_id=chain_id if isinstance(chain_id, int) else chain_id.value,
```

Comment:
i'd do it another way around - check if it's an enum, and only then get `.value`